source/howto/data/data-warehouse/redshift.rst

```diff
@@ -0,0 +1,177 @@
+.. meta::
+    :description: This how to guide shows you how to authenticate and run queries on Amazon RedShift.
+
+.. _howto-data-redshift:
+
+Access Redshift from an execution
+################################################
+
+.. admonition:: Prequirements
+    :class: warning
+
+    This guide assumes that you've:
+
+    * already created a Redshift cluster and have your data imported
+    * reviewed the clusters security group to allow connections from the security group ``valohai-sg-workers`` 
+
+You can access Amazon Redshift from your Valohai executions to run queries.
+
+
+Option 1) Allow ``ValohaiWorkerRole`` role to access your 
+-------------------------------------------------------------------
+
+Your AWS account has a IAM Role ``ValohaiWorkerRole``. This role is assigned to all EC2 instances that Valohai spins up to execute machine learning jobs.
+
+You can edit the permissions of that role inside your AWS subscription to give it access for Redshift.
+
+
+.. code-block:: json
+
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "VisualEditor0",
+                "Effect": "Allow",
+                "Action": "redshift:GetClusterCredentials",
+                "Resource": [
+                    "arn:aws:redshift:<region>:<account-ID>:dbuser:<cluster-identifier>/<username>",
+                    "arn:aws:redshift:*:<account-ID>:dbgroup:*/*",
+                    "arn:aws:redshift:<region>:<account-ID>:dbname:<cluster-identifier>/<database>"
+                ]
+            }
+        ]
+    }
+
+..
+
+.. tip::
+
+    You can create multiple roles and have Valohai environments that are connected to different roles.
+    
+    For example creating a ``ValohaiWorkerRole-Redshift`` that will be used only by certain Valohai environments, and those can be restricted to only certain teams in your Valohai organization.
+
+..
+
+You can use for example `redshift_connector <https://github.com/aws/amazon-redshift-python-driver>`_  connect to Redshift from your execution.
+
+Below a simple example:
+
+.. code-block:: python
+
+    import redshift_connector
+    import requests
+    import json
+
+    # Fetch credentials from the machines ValohaiWorkerRole
+    response = requests.get('http://169.254.169.254/latest/meta-data/iam/security-credentials/ValohaiWorkerRole')
+
+    # Parse the JSON results
+    credentials = json.loads(response.text)
+
+    # Connect to Redshift cluster using AWS credentials
+    conn = redshift_connector.connect(
+        iam=True,
+        host='<cluster-identifier>.xxxxxxxxx.xx-xxxx-x.redshift.amazonaws.com',
```

Comment:
All the "please fill in" parts could be defined separately above.

E.g.
```python
# Fill in your details to these variables

host='<cluster-identifier>...'
database='XXX'
db_user='XXX'
```

And then used below, i.e.
```python
conn = redshift_connector.connect(
    iam=True,
    host=host,
    database=database,
...
```